store/store.go

```diff
@@ -0,0 +1,377 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	eds "github.com/celestiaorg/celestia-node/share/new_eds"
+	"github.com/celestiaorg/celestia-node/store/cache"
+	"github.com/celestiaorg/celestia-node/store/file"
+)
+
+var (
+	log = logging.Logger("share/eds")
+
+	emptyAccessor = &eds.Rsmt2D{ExtendedDataSquare: share.EmptyExtendedDataSquare()}
+)
+
+const (
+	blocksPath  = "/blocks/"
+	heightsPath = blocksPath + "heights/"
+
+	defaultDirPerm = 0o755
+)
+
+var ErrNotFound = errors.New("eds not found in store")
+
+// Store is a storage for EDS files. It persists EDS files on disk in form of Q1Q4 files or ODS
+// files. It provides methods to put, get and remove EDS files. It has two caches: recent eds cache
+// and availability cache. Recent eds cache is used to cache recent blocks. Availability cache is
+// used to cache blocks that are accessed by sample requests. Store is thread-safe.
+type Store struct {
```

Comment:
For archival support we will eventually need a method that converts Q1Q4 into ODS

Reply:
We already support archival, but it will be part of archival pruning. It is planned as part of hardening.